developers/tutorials/ipfs-filecoin/using-nftstorage.md

```diff
@@ -0,0 +1,451 @@
+   
+       //2. Mint a NFT token on Harmony
+       const mintNFTTx = await sendTxToHarmony(metaData);
+   }
+   ```
+   
+3. **Upload NFT asset via NFT.Storage**
+
+   Now let's set up NFT.Storage to upload NFT assets.
+
+   + Install NFT.Storage dependency in your react app.
+
+     ```shell
+     npm install nft.storage
+     ```
+
+   + We are all set to start coding your logic, we need to first import NFTstorage and get the API Key that we created in the first step.
+
+     ```javascript
+     //Add this in MintNFT.js
+     import { NFTStorage } from "nft.storage";
+     const APIKEY = '<Your-API-KEY>';
+     ...
+     ```
+
+   + Then initializing NFTStorage connection using your API KEY.
+
+     ```javascript
+     const uploadNFTContent = async(inputFile) =>{
+     		//Initialize NFTStorage
+         const nftStorage = new NFTStorage({token: APIKEY,});
+     }
+     ```
+
+   + We can simply call `nftStorage.store()` to upload NFT content, which will upload your NFT as well as its related metadata to IPFS & Filecoin. You can customized your NFT metadata by updating properties like name and description.
+
+     ```javascript
+     const uploadNFTContent = async(inputFile) =>{
+         //Initialize NFTStorage
+         const nftStorage = new NFTStorage({token: APIKEY,});
+         try {
+             //Upload NFT to IPFS & Filecoin
+             const metadata = await nftStorage.store({
+                 name: 'Harmony NFT collection',
+                 description: 'This is a Harmony NFT collenction stored on IPFS & Filecoin.',
+                 image: file
+             });
+             return metadata;
+     
+         } catch (error) {
+             setErrorMessage("Could not save NFT to NFT.Storage - Aborted minting.");
+             console.log(error);
+         }
+     }
+     ```
+   
+     If NFT asset is uploaded successfully, NFT.Storage will do the following two things:
+
+     + Store NFT image on IPFS & Filecoin.
+     + Update NFT metadata with IPFS Link to your NFT image, then store NFT metadata on IPFS & Filecoin.
+   
+4. **Mint a new NFT on Harmony**
+
+   After getting the Metadata on IPFS, we can invoke the NFT smart contract on Harmony to mint a new NFT using NFT metadata on IPFS as tokenURI.
+
+   In this step, we will need to use `ethers` to interact with Metamask, so we first install this dependency.
+
+   ```shell
+   npm install ethers
+   ```
+
+   In order to invoke the NFT contract we have deployed to Harmony Testnet, we need to grab the contract ABI from the manifest JSON file which was generated via smart contract compilation.
+
+   ```shell
+   //run this command under harmony-nft folder
+   cp ../build/contracts/HarmonyNFT.json src/abi/
+   ```
+
+   Then we are ready to write code to send MintNFT transaction in `MintNFT.js` . First import ethers and smart contract manifest file,  and aslo create a `nftContractAddress` for the deployed NFT contract. 
```

Comment:
```suggestion
   Then we are ready to write code to send MintNFT transaction in `MintNFT.js` . This will import ethers and the smart contract manifest file, and also create a `nftContractAddress` for the deployed NFT contract. 
```